src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Touch/PlatformTouchEffect.ios.cs

```diff
@@ -247,21 +247,24 @@ public async Task HandleTouch(TouchStatus status, TouchInteractionStatus? intera
 			defaultShadowRadius = (float?)(defaultShadowRadius ?? renderer.Layer.ShadowRadius);
 			defaultShadowOpacity ??= renderer.Layer.ShadowOpacity;
 
-			await UIView.AnimateAsync(.2, () =>
-			{
-				if (color == Color.Default)
-					renderer.Layer.Opacity = isStarted ? 0.5f : (float)control.Opacity;
-				else
-					renderer.Layer.BackgroundColor = (isStarted ? color : control.BackgroundColor).ToCGColor();
-
-				renderer.Layer.CornerRadius = isStarted ? radius : defaultRadius.GetValueOrDefault();
-
-				if (shadowRadius >= 0)
+			var tcs = new TaskCompletionSource<UIViewAnimatingPosition>();
+			UIViewPropertyAnimator.CreateRunningPropertyAnimator(.2, 0, UIViewAnimationOptions.AllowUserInteraction,
+				() =>
 				{
-					renderer.Layer.ShadowRadius = isStarted ? shadowRadius : defaultShadowRadius.GetValueOrDefault();
-					renderer.Layer.ShadowOpacity = isStarted ? 0.7f : defaultShadowOpacity.GetValueOrDefault();
-				}
-			});
+					if (color == Color.Default)
+						renderer.Layer.Opacity = isStarted ? 0.5f : (float)control.Opacity;
+					else
+						renderer.Layer.BackgroundColor = (isStarted ? color : control.BackgroundColor).ToCGColor();
+
+					renderer.Layer.CornerRadius = isStarted ? radius : defaultRadius.GetValueOrDefault();
+
+					if (shadowRadius >= 0)
+					{
+						renderer.Layer.ShadowRadius = isStarted ? shadowRadius : defaultShadowRadius.GetValueOrDefault();
+						renderer.Layer.ShadowOpacity = isStarted ? 0.7f : defaultShadowOpacity.GetValueOrDefault();
+					}
+				}, endPos => tcs.SetResult(endPos));
```

Comment:
we need to check if task has already completed to prevent
![image](https://user-images.githubusercontent.com/33021114/150958195-42b53601-efa2-4b84-b21a-eedb1f7e9fd0.png)